### PR TITLE
[WEBSITE-472][WEBSITE-452] Global search and status--banner implement…

### PR DESCRIPTION
**Pull request by UsmanAfzal**
…ation

### CODE REVIEW

File: src/elements/_form.pug [hunk 1]
```diff
@@ -24,7 +24,7 @@ mixin input(type, label, name, id, datatype, datasize)
     else
       .input-group
         input(type=type, id=id, name=name)&attributes(attributes)
-        label(for=id)= label
+        label(for=id)&attributes(attributes)= label
```
Comment (naseberry) [inline src/elements/_form.pug]: Input and label will have the same attribute,

File: src/javascripts/search.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    searchToggle = header.querySelector('.search__global__toggle');
+
+  if(searchComponent) {
+    checkScreenSize();
+    searchToggle.addEventListener('click', toggleSearch);
+    window.addEventListener('orientationchange', checkScreenSize);
+  }
+
+  function checkScreenSize() {
+    if (window.innerWidth <= 768) {
```
Comment (naseberry) [inline src/javascripts/search.js]: should value '768' be stored as a variable?

File: src/javascripts/search.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    header = document.querySelector('header'),
+    logo = header.querySelector('.uk_parliament'),
+    searchComponent = header.querySelector('.search__global'),
+    input = header.querySelector('input'),
+    button = header.querySelector('button'),
+    searchToggle = header.querySelector('.search__global__toggle');
+
+  if(searchComponent) {
+    checkScreenSize();
+    searchToggle.addEventListener('click', toggleSearch);
```
Comment (naseberry) [inline src/javascripts/search.js]: on click, search.svg spin, check css transformation of svg's

File: src/stylesheets/elements/_links.scss [hunk 4]
```diff
@@ -156,3 +156,77 @@ footer a  {
...
+  .search__global__toggle {
+    display: none;
+  }
+}
+
+@media (max-width: $desktop - 1) {
+  .search__global__toggle {
+    top: 0;
+    right: 0;
+    width: 60px;
```
Comment (naseberry) [inline src/stylesheets/elements/_links.scss]: I thought fixed widths were frowned upon?

File: src/templates/prototypes/_buttons.pug [hunk 5]
```diff
@@ -45,14 +55,15 @@ block content
         +form('javascript:console.log("search submitted")', 'post')
           .input-group
             +input('search', 'Enter a keyword', 'q', 'search_box')(maxlength='1000', pattern='^(?!.*<script|<Script).*$')
-            //- button [value, style, type, datatype]
-            +button('Search', 'btn--primary', 'submit', 'search')
+            button(type='search', class='btn--white', data-tracking='search')
+              include ../../icons/download.svg
```
Comment (naseberry) [inline src/templates/prototypes/_buttons.pug]: why 'download.svg' for search components?

File: src/templates/prototypes/_buttons.pug [hunk 6]
```diff
@@ -45,14 +55,15 @@ block content
...
 
         h3 Find your MP and constituency
         +form('javascript:console.log("postcode submitted")', 'post')
           .input-group
             +input('text', 'Enter your full postcode, for example SW1A 0AA.', 'postcode_main', 'postcodeSearch')(maxlength='8', pattern='[0-9a-zA-Z ]{5,}')
             //- button [value, style, type, datatype]
-            +button('Find', 'btn--primary', 'submit', 'postcode')
+            button(type='search', class='btn--white', data-tracking='search')
+              include ../../icons/download.svg
```
Comment (naseberry) [inline src/templates/prototypes/_buttons.pug]: same as above.

File: src/stylesheets/elements/_links.scss (hunk 4 above)
Comment (UsmanAfzal) [reply src/stylesheets/elements/_links.scss]: Fixed heights yeh, not really widths. Plus this is an absolute within an absolute and also as each icon will have varying widths, its needs to be specified to match the size of the input-group icon.

File: src/javascripts/search.js (hunk 3 above)
Comment (UsmanAfzal) [reply src/javascripts/search.js]: Disabling the invocation of buttonloader as we no longer require iy. Not deleting the js file for now

File: src/javascripts/search.js (hunk 2 above)
Comment (UsmanAfzal) [reply src/javascripts/search.js]: yep adding

File: src/elements/_form.pug (hunk 1 above)
Comment (vinternet-pds) [reply src/elements/_form.pug]: Did we discuss passing a class to the into the mixin which would be applied to the label only?

Comment (UsmanAfzal) [reply src/elements/_form.pug]: Wasn't supposed to be added, removing this.

File: src/templates/prototypes/_buttons.pug (hunk 5 above)
Comment (vinternet-pds) [reply src/templates/prototypes/_buttons.pug]: If the download arrow icon works for both the search and download buttons then that's fine with me (I suppose we could rename it to something more suitable).

Comment (UsmanAfzal) [reply src/templates/prototypes/_buttons.pug]: Wrong one here, forgot to update once we got the icon.

File: src/elements/_form.pug (hunk 1 above)
Comment (UsmanAfzal) [reply src/elements/_form.pug]: I remember James you were going to make amends here

File: src/stylesheets/elements/_form.scss [hunk 7]
```diff
@@ -232,9 +233,13 @@ button:-moz-focusring,
 .input-group, .input-group__verbose {
   display: flex;
   flex-wrap: wrap;
+  flex-direction: row;
+  justify-content: flex-end;
```
Comment (naseberry) [inline src/stylesheets/elements/_form.scss]: radio button, check boxes and selects have now been affect, their styling is off.

File: src/templates/prototypes/_buttons.pug [hunk 8]
```diff
@@ -45,14 +55,15 @@ block content
         +form('javascript:console.log("search submitted")', 'post')
           .input-group
             +input('search', 'Enter a keyword', 'q', 'search_box')(maxlength='1000', pattern='^(?!.*<script|<Script).*$')
-            //- button [value, style, type, datatype]
-            +button('Search', 'btn--primary', 'submit', 'search')
+            button(type='search', class='btn--white', data-tracking='search')
+              include ../../icons/search.svg
```
Comment (naseberry) [inline src/templates/prototypes/_buttons.pug]: displaying the download image.

File: src/stylesheets/elements/_form.scss (hunk 7 above)
Comment (UsmanAfzal) [reply src/stylesheets/elements/_form.scss]: updated

**Pull request merged**